src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -253,6 +259,43 @@ public interface IPropertyReferenceExpression : IMemberReferenceExpression
     }
 
     /// <summary>
+    /// Represents a reference to an event.
+    /// </summary>
+    public interface IEventReferenceExpression : IMemberReferenceExpression
+    {
+        /// <summary>
+        /// Referenced event.
+        /// </summary>
+        IEventSymbol Event { get; }
+    }
+
+    /// <summary>
+    /// Represents a binding of an event.
+    /// </summary>
+    public interface IEventAssignmentExpression : IExpression
+    {
+        /// <summary>
+        /// Event being bound.
+        /// </summary>
+        IEventSymbol Event { get; }
+
+        /// <summary>
+        /// Instance used to refer to the event being bound.
+        /// </summary>
+        IExpression EventInstance { get; }
+
+        /// <summary>
+        /// Handler supplied for the event.
+        /// </summary>
+        IExpression HandlerValue { get; }
+
+        /// <summary>
+        /// True for adding a binding, false for removing one.
+        /// </summary>
+        bool Adds { get; }
```

Comment:
I had the same reaction. Maybe an enum is OK. This API just apes the bound trees.